lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -271,11 +312,89 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         this.categoryControlVisible = false;
     }
 
-    cancelChanges() {
+    cancelChanges(event: Event) {
+        event.stopPropagation();
         this.revertChanges();
         this.loadProperties(this.node);
     }
 
+    cancelEditChanges(buttonType: string, event: Event, group?: any) {
+        this.cancelChanges(event);
+        
+        switch (buttonType) {
+            case 'generalInfo':
+                this.editable = !this.editable;
+                break;
+            case 'tags':
+                this.editableTags = !this.editableTags;
+                break;
+            case 'categories':
+                this.editableCategories = !this.editableCategories;
+                break;
+            case 'group':
+                if (group) {
+                    group.editable = !group.editable;
+                }
+                break;
+            
+            default:
+                break;
+        }
+    }
+
+    toggleGeneralEdit(event: Event): void {
+        event.stopPropagation();
+        this.editable = !this.editable;
+        if (this.editable) {
+            this.panel.open();
+        }
+    }
+
+    toggleTagsEdit(event: Event): void {
+        event.stopPropagation();
+        this.editableTags = !this.editableTags;
+        this.tagNameControlVisible = true;
+        if (this.editableTags) {
```

Comment:
It can be just: 
this.tagsPanelState = this.editableTags 

because when editableTags is true then tagsPanelState is true too and the same for false